clientv3/client.go

```diff
@@ -128,14 +132,23 @@ func (c *Client) SetEndpoints(eps ...string) {
 	c.mu.Lock()
 	c.cfg.Endpoints = eps
 	c.mu.Unlock()
-	c.balancer.UpdateAddrs(eps...)
 
-	if c.balancer.NeedUpdate() {
+	var addrs []resolver.Address
+	for _, ep := range eps {
+		addrs = append(addrs, resolver.Address{Addr: ep})
+	}
+
+	c.resolver.NewAddress(addrs)
+	// TODO: Is this still needed now that we have the endpoints resolver?
+	c.balancer.UpdateAddrs(addrs)
```

Comment:
Yeah seems we don't need `UpdateAddrs` anymore. 

Given set1=[A,B,C] and set2=[D,E,F], sub-connections in set1 would be closed after `NewAddress`?

Reply:
Yes, it appears grpc manages this for us. I would like to add some test coverage to ensure it, but based on the zap log output, the connections are being closed when we the addr list via `NewAddress`.